src/vs/workbench/services/backup/node/backupFileService.ts

```diff
@@ -147,6 +158,10 @@ export class BackupFileService implements IBackupFileService {
 				return void 0; // return early if backup version id matches requested one
 			}
 
+			if (resource.scheme === 'file') {
+				content = `${resource.fsPath}\n${content}`;
```

Comment:
@Tyriar I suggest to use `Uri.toString()` here and `Uri.parse()` on the other side to reconstruct.